code/data_structures/src/splay_tree/splay.cpp

```diff
@@ -0,0 +1,283 @@
+#include <iostream>
+#include <stdlib.h>
+
+struct Node
+{
+    int value;
+    Node *left;
+    Node *right;
+    Node *parent;
+};
+Node *g_root;
+
+void inOrder(Node *cur, std::string &s)
```

Comment:
All these functions operate on a node. Thus, it would be better to use a class instead.